docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,214 @@
+
+Usual **devices** are **PLCs** (**P**rogrammable **L**ogic **C**ontrollers), **IPCs** (**I**ndustrial **PC**s) or any kind of **SoC-based** or **Microcontroller-based** Embedded System. The figure below shows a simplified conceptual model of such a device.
+
+![Simple Device Model](images/simple-device-model.svg)
+
+<!--
+* TODO: add somehow "Such a **device** is most often a specialized hardware that has been ruggedized to operate in harsh electrical, thermic and mechanical industrial environments."
+-->
+* The **Domain Application** is a program, that contains the domain specific process logic.
+  * On a **PLC** it's a _control application_, created by an _automation engineer_ in a domain specific graphical programming language (like "Ladder Diagram" or "Function block diagram")
+  * Or on a **SoC-based** or **Microcontroller-based** system it's an _application program_, created by an _embedded software engineer_ usually in C/C++
+* The **OS / Libs / Runtime** provide basic functionality to the **Domain Application**
+  <!-- TODO: add somehow "used to abtracts the hardware. But: on a microcontroler usually less abtraction, more hw dep on the domain app, and even no OS" -->
+* **Sensors** and **Actuators** enable the **device** to observe and control physical behaviour on the shopfloor or device's environment.
+  * can be integrated in the **device's** hardware,
+    or connected to the **device** via some **Fieldbus** (e.g. Modbus, CANopen, PROFINET, EtherCAT, ...) or
+    some **Local Interface** as USB, UART, SPI, I2C, ...
+  * can be simple peripherals as a light barrier, or a sensor for temperature or pressure;
+    or complex equipments as robotic arms or even other **devices**
+* **Inputs / Outputs** are the communication channels between the **Domain Application** and **Sensors** and **Actuators**
+  * drivers (as part of the **OS / Libs / Runtime** and/or the **Domain Application**) do expose all data from
+    **Sensors** and **Actuators** to the **device** as inputs or outputs
+  * also the **Domain Application** can expose data as input or output (e.g. own _signals_ or _states_)
+
+## IoT Cloud Domain
+
+Cloud end-points could be IoT Cloud Platforms, as e.g. [Cumulocity](https://www.softwareag.cloud/site/product/cumulocity-iot.html), [Azure](https://azure.microsoft.com) or [AWS](https://aws.amazon.com); or end-points for specific IoT tasks, as e.g. a time-series database to collect measurements.
+
+* Each **device** is represented in the IoT cloud by an individual **device twin**
+* All information / data or functionality the device provides to the cloud appears in context of it's **device twin**
+* The focus of **thin-edge** is on two IoT Cloud aspects:
+  - 1) **Telemetry Data Handling**
+  - 2) **Device Management**
+
+### Telemetry Data Handling
+
+**Telemetry data** are the measurements, events and alarms collected or raised by sensors, actuators or domain applications.
```

Comment:
IMHO simplify this by removing "sensors, actuators or domain applications." or putting it into a seperate sentence. Notes that sensors / actuators / domain applications are NOT part of thin-edge domain model, so its a bit irratating / too much information.

Reply:
Applied in new PR, see https://github.com/thin-edge/thin-edge.io/pull/1554#issuecomment-1301017645